---
title: "Supplemental File 3. Analysis of vaginal microbiota dynamics in relation to
  pubertal development"
author: "Roxana J. Hickey"
output:
  html_document:
    keep_md: yes
    toc: yes
  pdf_document:
    toc: yes
---

# Description
This is a supplement to the paper "Vaginal microbiota of adolescent girls resemble those of reproductive-age women prior to the onset of menarche" by Hickey et al. Please refer to the paper for complete information about the objectives and study design.

The embedded R code works through the first set of analyses and generation of figures related to the assessment of vaginal microbiota composition in girls and mothers. The analyses can be run directly from the R Markdown file "adolescent-supp-03.Rmd" using RStudio. It should be run after "adolescent-supp-01.Rmd" and "adolescent-supp-02.Rmd".

## Objective
The first set of analyses dealt with characterizing vaginal microbiota composition and identifying major groups using clustering and ordination approaches. Next, we will take a closer look at the dynamics of these communities over time as girls progress through puberty and menarche. Special attention is devoted to assessing trends in the relative abundance of lactic acid bacteria and vaginal pH, as these are generally considered indicators of a 'healthy' vaginal microbiota in reproductive age women.

***
# Initial setup

Clear the workspace, load data from the previous step, and load necessary packages.

*Note: If you run the R Markdown script 'as is' from the same directory containing it and the 'data-input' and 'data-postproc' subdirectories, all figures will be printed inside the resulting PDF or HTML output. If you want to save the figures as individual files, uncomment any lines throughout the script starting with 'ggsave()' or 'pdf()'. I made note of each of these within the chunk code.*

```{r setup}
## Clear current workspace
rm(list=ls())

## Load the RData file created from adolescent-supp-02.Rmd
## (this contains data from supp-01 as well)
load("data-postproc/02-cluster-pcoa-last-run.RData")

## Load packages
library(ape)
library(ggplot2)
library(gplots)
library(grid)
library(gridExtra)
library(reshape)
library(scales)

## Write new directories for output
dir.create("community-dynamics")
dir.create("regression-analysis")
```

***
# Part I: Summarize community dynamics of vaginal (and vulvar) microbiota

First we will create an individual summary for each individual participant (and her mother, if applicable) to look at community composition of the vaginal and vulvar microbiota, along with selected metadata, at every visit during the study. The output is a PDF file with each page containing the following plots for a given subject:

* Community composition of girl's vaginal microbiota
* Community composition of girl's vulvar microbiota
* Community composition of mother's vaginal microbiota
* Girl's menarcheal status
* Girl's Tanner breast and genital stages (clinician-assessed)
* Vaginal pH of girl and mother (when available)

## Supplemental File 4. Individual summaries of vaginal and vulvar microbiota composition and associated metadata for all study participants

```{r supp-4-summary-plots}
## g_legend function from http://stackoverflow.com/questions/11883844/
## inserting-a-table-under-the-legend-in-a-ggplot2-histogram
g_legend <- function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

pdf(file="community-dynamics/adolescent-supp-04.pdf", 
    width=8.5, height=11, pointsize=8)
for (i in as.numeric(unique(meta$gm.pair))){
  pick <- which(as.numeric(meta$gm.pair) == i)
  if (length(pick) >= 1){
    prop.tmp <- prop.red[,pick]
    meta.tmp <- meta[pick,]
    
    meta.tmp$race <- gsub("bla", "Black", meta.tmp$race)
    meta.tmp$race <- gsub("cauc", "Caucasian", meta.tmp$race)
    meta.tmp$race <- gsub("ami", "Native American", meta.tmp$race)
    meta.tmp$ethn <- gsub("nhis", "non-Hispanic", meta.tmp$ethn)
    meta.tmp$ethn <- gsub("hisp", "Hispanic", meta.tmp$ethn)
    
    taxa.pick25 <- order(rowMeans(as.matrix(prop.tmp), 
                                  na.rm=TRUE), 
                         decreasing=T)[1:25]
    prop.top25.tmp <- prop.tmp[taxa.pick25,]
    
    meta.prop25.tmp <- data.frame(cbind(meta.tmp$type.site, 
                                        meta.tmp$visit, 
                                        meta.tmp$men.stat,
                                        meta.tmp$tan.br.dr,
                                        meta.tmp$tan.gen.dr,
                                        meta.tmp$ph,
                                        t(prop.top25.tmp)))
    colnames(meta.prop25.tmp)[1:6] <- c("type.site", 
                                        "visit",
                                        "men.stat",
                                        "tan.br.dr",
                                        "tan.gen.dr",
                                        "ph")
    
    meta.prop25.lg.tmp <- melt(meta.prop25.tmp,
                               id.vars=c("type.site", 
                                         "visit",
                                         "men.stat",
                                         "tan.br.dr",
                                         "tan.gen.dr",
                                         "ph"))
    colnames(meta.prop25.lg.tmp)[7:8] <- c("taxon", "prop")
    
    meta.prop25.lg.tmp$visit <- as.integer(unfactor(meta.prop25.lg.tmp$visit))
    meta.prop25.lg.tmp$prop <- as.numeric(unfactor(meta.prop25.lg.tmp$prop))
    
    if (length(levels(meta.prop25.lg.tmp$type.site)) == 2){
      levels(meta.prop25.lg.tmp$type.site) <- c("Girl Vagina", 
                                                "Girl Vulva")}
    else if (length(levels(meta.prop25.lg.tmp$type.site)) == 3){
      levels(meta.prop25.lg.tmp$type.site) <- c("Girl Vagina", 
                                                "Girl Vulva", 
                                                "Mother Vagina")}
    
    ## Community composition
    gg.taxa <- ggplot(meta.prop25.lg.tmp, 
                      aes(x=visit, y=prop, fill=taxon))
    gg.taxa.bar <- gg.taxa + geom_bar(stat="identity") +
      facet_wrap(~ type.site, ncol=1, scale="fixed") +
      scale_fill_manual(values=col.taxa[taxa.pick25], name="Taxon") +
      scale_x_discrete(limits=seq(1, max(meta.prop25.lg.tmp$visit), by=1), 
                       name="") +
      ylab("Proportion") +
      ggtitle(paste(unique(meta.tmp$gm.pair.fullID), ": ",
                    unique(meta.tmp$race), ", ",
                    unique(meta.tmp$ethn), ", ", 
                    round(min(na.exclude(meta.tmp$age.sampling)), digits=1),
                    " y/o at enrollment\n", sep="")) +
      theme_cust +
      theme(axis.text.x=element_blank(),
            legend.text=element_text(size=6),
            legend.title=element_text(size=8),
            legend.position=c(0,0.5),
            legend.justification=c(0,0.5))
    
    ## Now we need to go back to the original metadata table which has each
    ## visit listed in its own row. This way we don't have duplicate menarche
    ## status and Tanner scores to plot (one matching vagina, one matching
    ## vulva when they are identical)
    meta2.tmp <- subset(meta.orig, 
                        sample.ID.vag %in% rownames(meta.tmp) | 
                          sample.ID.vul %in% rownames(meta.tmp))
    
    meta2.tmp$men.stat <- factor(meta2.tmp$men.stat, 
                                  levels=c(NULL,"post","pre"))
    
    ## Menarche status
    gg.men.stat <- ggplot(subset(meta2.tmp, type=="girl"), 
                          aes(x=visit, y=subject, color=men.stat))
    gg.men.stat.dot <- gg.men.stat + 
      geom_point(size=5) +  
      scale_color_manual(values=col.men.stat[2:3],
                         breaks=c("pre","post"),
                         labels=c("Pre","Post"),
                         na.value="gray70",
                         name="Menarche status",
                         drop=FALSE) +
      scale_x_discrete(limits=seq(1, max(meta.prop25.lg.tmp$visit), by=1), 
                       name="") +
      scale_y_discrete(labels="Menarche", name="") +
      theme_cust +
      theme(axis.text.x=element_blank(),
            legend.text=element_text(size=6),
            legend.title=element_text(size=8),
            legend.position=c(0,1),
            legend.justification=c(0,1))
    
    ## Tanner
    tanner.tmp <- meta2.tmp[,c("visit",
                               "tan.br.dr",
                               "tan.gen.dr",
                               "age.sampling")]
    
    tanner.lg.tmp <- melt(tanner.tmp, id.vars=c("visit", "age.sampling"))
    
    tanner.lg.tmp$value <- factor(tanner.lg.tmp$value, 
                                  levels=c("1","2","3","4","5"))
    
    gg.tanner <- ggplot(tanner.lg.tmp, 
                        aes(x=visit, y=variable, color=value))
    gg.tanner.dot <- gg.tanner +  
      geom_point(size=5) +  
      scale_color_manual(values=col.tanner,
                         name="Tanner stage",
                         drop=FALSE) +  
      scale_x_discrete(limits=seq(1, max(tanner.lg.tmp$visit), by=1),
                       name="Visit") +
      scale_y_discrete(breaks=c("tan.gen.dr","tan.br.dr"), 
                       labels=c("Genital", "Breast"), 
                       name="") +
      theme_cust +
      theme(legend.text=element_text(size=6),
            legend.title=element_text(size=8),
            legend.position=c(0,1),
            legend.justification=c(0,1))
    
    # pH
    gg.ph <- ggplot(subset(meta2.tmp, type=="girl"), 
                    aes(x=visit, y=ph))
    gg.ph.line <- gg.ph +
      geom_point(size=2) +
      geom_line(lty=3) +
      scale_x_discrete(limits=seq(1, max(meta2.tmp$visit), by=1), 
                       name="Visit") +
      scale_y_continuous(limits=c(4.0, 8.0)) +
      geom_text(label=meta2.tmp$ph[meta2.tmp$type=="girl"], 
                vjust=-0.5, size=3.5) +
      ylab("Vaginal pH") +
      theme_cust_nominor
    
    check.ph <- is.na(meta2.tmp$ph[meta2.tmp$type=="girl"]) 

    if  ("FALSE" %in% check.ph) {
      grid.arrange(gg.taxa.bar + theme(legend.position="none",
                                       plot.margin=unit(c(0.25, 0.5, 0.25, 2), "lines")),
                   g_legend(gg.taxa.bar), 
                   gg.men.stat.dot + theme(legend.position="none",
                                           plot.margin=unit(c(0.25, 0.5, 0.25, 0), "lines")),
                   g_legend(gg.men.stat.dot),
                   gg.tanner.dot + theme(legend.position="none",
                                         axis.text.x=element_blank(),
                                         axis.title.x=element_blank(),
                                         plot.margin=unit(c(0.25, 0.5, 0.25, 1), "lines")),
                   g_legend(gg.tanner.dot),
                   gg.ph.line + theme(legend.position="none",
                                      plot.margin=unit(c(0.25, 0.5, 0.25, 3.25), "lines")),
                   ncol=2, widths=c(4,1), heights=c(6, 1, 1.25, 1.25))
      }
    else 
      grid.arrange(gg.taxa.bar + theme(legend.position="none",
                                       plot.margin=unit(c(0.25, 0.5, 0.25, 2), "lines")),
                   g_legend(gg.taxa.bar), 
                   gg.men.stat.dot + theme(legend.position="none",
                                           plot.margin=unit(c(0.25, 0.5, 0.25, 0), "lines")),
                   g_legend(gg.men.stat.dot),
                   gg.tanner.dot + theme(legend.position="none",
                                         plot.margin=unit(c(0.25, 0.5, 0.25, 1), "lines")),
                   g_legend(gg.tanner.dot),
                   ncol=2, widths=c(4,1), heights=c(7, 1, 1.5))
    }
  }
dev.off()
```

## Highlight examples of _Lactobacillus_ dominant vaginal microbiota

Next we will plot just a few examples of vaginal microbiota with _Lactobacillus_ predominant by menarche. I have selected subjects 102, 103, 107 and 109. We will plot only the vaginal microbiota composition, menarche status, and Tanner breast stage for each. This also requires a bit of setup:

```{r fig-5-setup}
## Subset prop and meta for subjects 102, 103, 107 and 109
meta.fig5 <- subset(meta, subject %in% c(102, 103, 107, 109) & site=="vag")
prop.fig5 <- prop.red[,colnames(prop.red) %in% rownames(meta.fig5)]

taxa.pick40 <- order(rowMeans(as.matrix(prop.fig5), na.rm=TRUE),
                     decreasing=T)[1:40]
prop.fig5.top40 <- prop.fig5[taxa.pick40,]
    
mp.fig5 <- data.frame(cbind(as.character(meta.fig5$subject),
                            meta.fig5$visit, 
                            as.character(meta.fig5$men.stat),
                            meta.fig5$tan.br.dr,
                            t(prop.fig5.top40)))
colnames(mp.fig5)[1:4] <- c("subject",
                            "visit",
                            "men.stat",
                            "tan.br.dr")

mp.fig5.lg <- melt(mp.fig5, id.vars=c("subject",
                                      "visit",
                                      "men.stat",
                                      "tan.br.dr"))

mp.fig5.lg$visit <- as.integer(unfactor(mp.fig5.lg$visit))
mp.fig5.lg$value <- as.numeric(unfactor(mp.fig5.lg$value))

mp.fig5.lg$tan.br.dr <- factor(mp.fig5.lg$tan.br.dr, levels=c(1,2,3,4,5))

mp.s102 <- subset(mp.fig5.lg, subject==102)
mp.s103 <- subset(mp.fig5.lg, subject==103)
mp.s107 <- subset(mp.fig5.lg, subject==107)
mp.s109 <- subset(mp.fig5.lg, subject==109)

## Plot vaginal microbiota composition of subject 102 (w/ legend)
gg.taxa.bar.102 <- ggplot(mp.s102, aes(x=visit, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=col.taxa[taxa.pick40], name="Taxon") +
  scale_x_discrete(limits=seq(1, max(mp.s102$visit), by=1), 
                   name="") +
  ylab("Proportion") +
  ggtitle("Subject 102") +
  theme_cust +
  guides(fill=guide_legend(ncol=8)) +
  theme(axis.text.x=element_blank(),
        legend.text=element_text(size=6),
        legend.title=element_text(size=8),
        legend.position=c(0,0),
        legend.justification=c(0,0),
        legend.direction="horizontal")

## Plot vaginal microbiota composition of subject 103
gg.taxa.bar.103 <- ggplot(mp.s103, aes(x=visit, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=col.taxa[taxa.pick40], name="Taxon") +
  scale_x_discrete(limits=seq(1, max(mp.s103$visit), by=1), 
                   name="") +
  ylab("Proportion") +
  ggtitle("Subject 103") +
  theme_cust +
  theme(axis.text.x=element_blank(),
        legend.position="none")

## Plot vaginal microbiota composition of subject 107
gg.taxa.bar.107 <- ggplot(mp.s107, aes(x=visit, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=col.taxa[taxa.pick40], name="Taxon") +
  scale_x_discrete(limits=seq(1, max(mp.s107$visit), by=1), 
                   name="") +
  ylab("Proportion") +
  ggtitle("Subject 107") +
  theme_cust +
  theme(axis.text.x=element_blank(),
       legend.position="none")

## Plot vaginal microbiota composition of subject 109
gg.taxa.bar.109 <- ggplot(mp.s109, aes(x=visit, y=value, fill=variable)) +
  geom_bar(stat="identity") +
  scale_fill_manual(values=col.taxa[taxa.pick40], name="Taxon") +
  scale_x_discrete(limits=seq(1, max(mp.s109$visit), by=1), 
                   name="") +
  ylab("Proportion") +
  ggtitle("Subject 109") +
  theme_cust +
  theme(axis.text.x=element_blank(),
        legend.position="none")

## Plot Tanner/menarche of subject 102 (w/ legend)
gg.tb.dot.102 <- ggplot(mp.s102, aes(x=visit, y=subject, 
                                     shape=men.stat, color=tan.br.dr)) + 
  geom_point(size=5) +  
  scale_color_manual(values=col.tanner,
                     name="Tanner breast",
                     drop=FALSE) +
  scale_shape_manual(breaks=c("pre", "post"),
                     labels=c("Pre", "Post"),
                     values=c(16,1),
                     name="Menarche status") +
  scale_x_discrete(limits=seq(1, max(mp.s102$visit), by=1), 
                   name="Visit") +
  ylab("") +
  theme_cust +
  theme(axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        legend.text=element_text(size=6),
        legend.title=element_text(size=8),
        legend.position=c(0.7,0.5),
        legend.justification=c(0.5,0.5),
        legend.direction="horizontal")

## Plot Tanner/menarche of subject 103
gg.tb.dot.103 <- ggplot(mp.s103, aes(x=visit, y=subject, 
                                     shape=men.stat, color=tan.br.dr)) + 
  geom_point(size=5) +  
  scale_color_manual(values=col.tanner,
                     name="Tanner breast",
                     drop=FALSE) +
  scale_shape_manual(breaks=c("pre", "post"),
                     labels=c("Pre", "Post"),
                     values=c(16,1),
                     name="Menarche status") +
  scale_x_discrete(limits=seq(1, max(mp.s103$visit), by=1), 
                   name="Visit") +
  ylab("") +
  theme_cust +
  theme(axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        legend.position="none")

## Plot Tanner/menarche of subject 107
gg.tb.dot.107 <- ggplot(mp.s107, aes(x=visit, y=subject, 
                                     shape=men.stat, color=tan.br.dr)) + 
  geom_point(size=5) +  
  scale_color_manual(values=col.tanner,
                     name="Tanner breast",
                     drop=FALSE) +
  scale_shape_manual(breaks=c("pre", "post"),
                     labels=c("Pre", "Post"),
                     values=c(16,1),
                     name="Menarche status") +
  scale_x_discrete(limits=seq(1, max(mp.s107$visit), by=1), 
                   name="Visit") +
  ylab("") +
  theme_cust +
  theme(axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        legend.position="none")

## Plot Tanner/menarche of subject 109
gg.tb.dot.109 <- ggplot(mp.s109, aes(x=visit, y=subject, 
                                     shape=men.stat, color=tan.br.dr)) + 
  geom_point(size=5) +  
  scale_color_manual(values=col.tanner,
                     name="Tanner breast",
                     drop=FALSE) +
  scale_shape_manual(breaks=c("pre", "post"),
                     labels=c("Pre", "Post"),
                     values=c(16,1),
                     name="Menarche status") +
  scale_x_discrete(limits=seq(1, max(mp.s109$visit), by=1), 
                   name="Visit") +
  ylab("") +
  theme_cust +
  theme(axis.ticks.y=element_blank(),
        axis.text.y=element_blank(),
        legend.position="none")
```

## Figure 5. Transitions to _Lactobacillus_-dominant vaginal microbiota in four perimenarcheal girls.
Each panel shows the vaginal bacterial community profiles and associated pubertal development of four participants sampled longitudinally. The top bar plot of each panel shows the proportions of bacterial taxa present at each sampling, with colors of taxa indicated in the legend at bottom left. Below each bar plot the menarcheal status (M), Tanner breast (TB) and Tanner genital (TG) scores as assessed by a clinician are indicated following the color scheme shown in the legend at bottom right. The participant’s age in years at the time of sampling is indicated below the bars in each panel. Empty spaces in the plots indicate a skipped scheduled quarterly visit or a visit in which a vaginal swab or metadata were not collected.

```{r fig-5-community-dynamics, fig.width=12, fig.height=8}
## Uncomment next line to save as a PDF
# pdf("community-dynamics/fig-5-community-dynamics.pdf", width=12, height=8, pointsize=8)
grid.arrange(gg.taxa.bar.107 + theme(plot.margin=unit(c(0.25, 1, 0, 0.8), "lines")),
             gg.taxa.bar.109 + theme(plot.margin=unit(c(0.25, 1, 0, 0.8), "lines")),
             gg.tb.dot.107 + theme(plot.margin=unit(c(0.25, 1, 0.5, 2.4), "lines")),
             gg.tb.dot.109 + theme(plot.margin=unit(c(0.25, 1, 0.5, 2.4), "lines")),
             gg.taxa.bar.102 + theme(legend.position="none",
                                     plot.margin=unit(c(0.25, 1, 0, 0.8), "lines")),
             gg.taxa.bar.103 + theme(plot.margin=unit(c(0.25, 1, 0, 0.8), "lines")),
             gg.tb.dot.102 + theme(legend.position="none",
                                   plot.margin=unit(c(0.25, 1, 0.5, 2.4), "lines")),
             gg.tb.dot.103 + theme(plot.margin=unit(c(0.25, 1, 0.5, 2.4), "lines")),
             g_legend(gg.taxa.bar.102),
             g_legend(gg.tb.dot.102),
             ncol=2, heights=c(3,0.75,3,0.75,1.25))
dev.off()
```

## _Gardnerella_ in perimenarcheal vaginal microbiota

_Gardnerella vaginalis_ was surprisingly common among girls in our study. This is notable because the species (or genus) is commonly associated with bacterial vaginosis, and some have argued it is acquired through sexual contact. However, the girls in this study had no history of sexual activity and reported themselves in good health. Below we plot the changes in proportion of _Gardnerella_ over time in girls who had at least 5% _Gardnerella_ in her vaginal microbiota at any point in time during the study.

```{r gardnerella-vag-setup}
meta.vag$Gvag <- spe.prop.vag[,"Gardnerella_vaginalis"]
meta.vag$Gardnerella <- rowSums(spe.prop.vag[,grep("Gardnerella", colnames(spe.prop.vag))])

meta.vag.gard.05 <- subset(meta.vag, type=="girl" & Gardnerella >= 0.05)
unique(meta.vag.gard.05$subject) # n=11

## We want to plot Gardnerella over time in all of the girls who carried it at some point, 
## so we need to grab all the obervations for the girls listed above.
meta.vag.gard.05.fullobs <- meta.vag[meta.vag$subject %in% meta.vag.gard.05$subject,]
```

## Figure S3. Proportion of _Gardnerella_ over time in the vaginal microbiota of girls.
_Gardnerella_ was present in the vaginal microbiota at a proportion of 0.05 or greater at least once in 11/31 adolescent participants. Each subplot shows the proportion of _Gardnerella_ (encompassing sequence reads assigned to either the species level as _G. vaginalis_ or genus level as _Gardnerella_) in the vaginal microbiota of a single participant at each clinical visit. Open circles represent premenarcheal samples, and filled circles represent postmenarcheal samples. The x-axis indicates the clinical visit at which each sample was collected; visits occurred approximately every three months.

```{r fig-s3-gardnerella-vag, fig.width=8, fig.height=4}
gg.girl.gard <- ggplot(meta.vag.gard.05.fullobs, 
                       aes(x=visit, y=Gardnerella, group=subject, 
                           color=type, shape=men.stat))
gg.girl.gard + geom_line(lty=3, alpha=0.5) +
  facet_wrap( ~ gm.pair.fullID, ncol=4) +
  geom_point(size=3) +
  scale_color_manual(values="#01665E", guide=FALSE) +
  scale_shape_manual(values=c(16, 1), 
                     breaks=c("pre", "post"), 
                     labels=c("Pre", "Post"),
                     name="Menarche\nStatus", na.value=10) +
  xlab("Visit") +
  ylab("Proportion of Gardnerella") +
  scale_x_discrete(1:14, name="Visit No.") +
  ylim(0,1) +
  theme_cust_nominor +
  theme(axis.text=element_text(size=6),
        legend.justification=c(1,1), 
        legend.position=c(0.95,0.3))

## Uncomment next line to save as a PDF
# ggsave("community-dynamics/fig-s3-gardnerella-vag.pdf", width=8, height=4, units="in")
```

## _Gardnerella_ in vaginal and vulvar microbiota

I was also curious to see whether the same patterns were observed in the vulva. Not surprisingly, they are fairly similar, although subject 135 is an interesting exception with a might higher proportion of _Gardnerella_ on the vulva compared to the vagina.

```{r gardnerella-vag-vul-setup}
meta$Gvag <- t(prop.red["Gardnerella_vaginalis",])
meta$Gardnerella <- colSums(prop.red[grep("Gardnerella", rownames(prop.red)),])

meta.girl.gard.05 <- subset(meta, type=="girl" & Gardnerella >= 0.05)
unique(meta.girl.gard.05$subject) # n=11

## Subjects who had Gardnerella in vagina
unique(meta.girl.gard.05$subject[meta.girl.gard.05$site=="vag"]) # n=11

## Subjects who had Gardnerella on vulva
unique(meta.girl.gard.05$subject[meta.girl.gard.05$site=="vul"]) # n=10

meta.girl.gard.05.fullobs <- meta[meta$subject %in% meta.girl.gard.05$subject,]
```

Plot vaginal and vulvar samples of girls with at least 5% _Gardnerella_:
```{r gardnerella-vag-vul, fig.width=8, fig.height=4}
gg.girl.gard <- ggplot(meta.girl.gard.05.fullobs, 
                       aes(x=visit, y=Gardnerella, group=site, color=site, shape=men.stat))
gg.girl.gard + geom_line(lty=3, alpha=0.5) +
  facet_wrap( ~ gm.pair.fullID, ncol=4) +
  geom_point(size=3) +
  scale_color_manual(values=col.site, name="Site", 
                     breaks=c("vag","vul"), 
                     labels=c("Vagina", "Vulva")) +
  scale_shape_manual(values=c(16, 1), name="Menarche\nStatus", 
                     breaks=c("pre", "post"),
                     labels=c("Pre", "Post")) +
  xlab("Visit") +
  ylab("Proportion of Gardnerella") +
  scale_x_discrete(1:14, name="Visit No.") +
  ylim(0,1) +
  theme_cust_nominor +
  theme(axis.text=element_text(size=6))
```

***
# Part II: Multiple linear regression analysis of lactic acid bacteria and vaginal pH

Now we want to extract some general trends from the data. We are going to focus on trends in lactic acid bacteria and vaginal pH. We will use a fairly simple multiple linear regression approach to identify variables significantly related to these changes. First we should simplify our variables and select only one of the Tanner measures to use as a proxy for pubertal development (we currently have four: breast self-assessement, breast clinician-assessment, genital self-assessment, genital clinician assessment). To do so we simply calculate the correlations between Tanner scores by body site (breast/genital) and self/clinician assessment to see how well they agree.

## Correlation of Tanner scores

```{r tanner-correlations-setup}
## Spearman
cor.sp <- cor(meta[meta$type=="girl" & meta$site=="vag", c("tan.gen.dr","tan.gen.self","tan.br.dr","tan.br.self")], 
              use="pairwise.complete.obs", method="spearman")
cor.sp
```

## Figure S4. Correlations of Tanner breast and genital scores according to self and clinician assessment.
Spearman’s correlation coefficients were calculated to determine how well breast/genital scores agreed for both clinician- (upper left) and self-assessment (upper right).  Clinician- and self-assessments for Tanner breast (lower left) and Tanner genital (lower right) scores were also correlated. Correlation coefficients are reported in the header of each plot. Clinician-assessed Tanner breast scores were primarily used for analyses described in the paper.

```{r fig-s4-tanner-correlations, fig.width=10, fig.height=10}
## Tanner genital vs. Tanner breast (clinician)
gg.tan.dr <- ggplot(meta[meta$type=="girl" & meta$site=="vag",], aes(x=tan.br.dr, y=tan.gen.dr)) + 
  geom_jitter(position=position_jitter(width=0.2, height=0.2), color="#8B1C62", size=3, alpha=0.7) +
  ggtitle("Clinician Assessment\n(Spearman's rho = 0.73)") +
  xlab("Tanner Breast") +
  ylab("Tanner Genital") +
  theme_cust_nominor

## Tanner genital vs. Tanner breast (self)
gg.tan.self <- ggplot(meta[meta$type=="girl" & meta$site=="vag",], aes(x=tan.br.self, y=tan.gen.self)) + 
  geom_jitter(position=position_jitter(width=0.2, height=0.2), color="#8B1C62", size=3, alpha=0.7) +
  ggtitle("Self Assessment\n(Spearman's rho = 0.71)") +
  xlab("Tanner Breast") +
  ylab("Tanner Genital") +
  theme_cust_nominor

## Tanner breast (self) vs. Tanner breast (dr)
gg.tb.dr.self <- ggplot(meta[meta$type=="girl" & meta$site=="vag",], aes(x=tan.br.dr, y=tan.br.self)) + 
  geom_jitter(position=position_jitter(width=0.2, height=0.2), color="#8B1C62", size=3, alpha=0.7) +
  ggtitle("Tanner Breast, Self vs. Clinician\n(Spearman's rho = 0.63)") +
  xlab("Clinician") +
  ylab("Self") +
  theme_cust_nominor

## Tanner genital (self) vs. Tanner genital (dr)
gg.tg.dr.self <- ggplot(meta[meta$type=="girl" & meta$site=="vag",], aes(x=tan.gen.dr, y=tan.gen.self)) + 
  geom_jitter(position=position_jitter(width=0.2, height=0.2), color="#8B1C62", size=3, alpha=0.7) +
  ggtitle("Tanner Genital, Self vs. Clinician\n(Spearman's rho = 0.68)") +
  xlab("Clinician") +
  ylab("Self") +
  theme_cust_nominor

## Uncomment the next line to save as a PDF
# pdf("fig-s4-tanner-correlations.pdf", width=10, height=10)
multiplot(gg.tan.dr, gg.tan.self, gg.tb.dr.self, gg.tg.dr.self, layout=matrix(c(1,2,3,4), ncol=2, byrow=TRUE))
dev.off()
```

As the plots above show, the Tanner scores (breast and genital, self and clinician) are fairly well correlated. This is not too surprising since the Tanner scores for breast and genital were designed to capture simultaneous stages of development. One of the inclusion criteria for participation in the study was Tanner breast of at least stage 2, so it makes intuitive sense to rely the Tanner breast score as a proxy for pubertal development in our study. Additionally, the clinician-assessed Tanner breast score has more complete observations (192 out of 198 vagina samples) than the clinician assessed Tanner genital score (186/198). Therefore, we'll just use the Tanner breast score for the subsequent analyses.

## Multiple linear regression
Our goal here is to determine whether the proportion of lactic acid bacteria and vaginal pH are significantly associated with Tanner stage, age and/or menarche status. We'll set up several regression models (for each of the two response variables) and select a model **HOW**

*Note: In order to perform linear regression on proportional data (e.g., LAB proportions), the data should be approximately normally distributed. However, our proportions of LAB are heavily skewed with many close to 1 and quite a few close to zero, with far less in the middle. Here we subject the data to a logit transform, log(y/(1-y)), described by Warton and Hui as a preferred method for ecological non-binomial data such as proportions. However, because we have 0's and 1's in our data matrix, these would result in some -Inf and Inf values that cause problems for the linear model. An ad hoc solution to this is to add a nominal value epsilon to the numerator and denominator during the logit transform. Because our data are more skewed toward proportions close to 1 (e.g., check this with hist(meta.vagLAB)), we take as epsilon the difference between 1 and the largest proportion less than 1. This is defined below as eps. For discussion and justification of this approach, see Warton and Hui (2011) The arcsine is asinine: the analysis of proportions in ecology. Ecology 92:3–10.*

### Setup for modeling LAB
```{r fit-LAB-setup}
## First we add the proportions of individual Lactobacillus spp., Lactobacillus genus, and lactic acid bacteria as additional variables to the metadata table
meta$Lactobacillus_crispatus <- t(prop.red["Lactobacillus_crispatus",])
meta$Lactobacillus_gasseri <- t(prop.red["Lactobacillus_gasseri",])
meta$Lactobacillus_iners <- t(prop.red["Lactobacillus_iners",])
meta$Lactobacillus_jensenii <- t(prop.red["Lactobacillus_jensenii",])
meta$Lactobacillus <- colSums(prop.red[grep("Lactobacillus", rownames(prop.red)),])
meta$LAB <- colSums(prop.red[c(6, 28, 43:49, 72:74),]) # 6=Aerococcus, 28=Facklamia, 43:49=Lactobacillus, 72:74=Streptococcus

## Define modified logit transform so we can specify epsilon adjustment
mod.logit <- function(x, eps) { log((x+eps)/(1-x+eps)) }

## Calculate epsilon:
## Subtract values of LAB from 1 
dif.lab <- 1-meta$LAB

## Take the smallest absolute difference greater than zero (~0.000075)
eps.lab <- min(dif.lab[dif.lab > 0])
eps.lab

## Add logit-transformed LAB proportions to metadata
meta$LAB.logit <- mod.logit(meta$LAB, eps.lab)

## Subset metadata to only girl vagina samples and exclude Tanner breast = 1 (only 3 samples)
meta.girl.vag <- subset(meta, type=="girl" & site=="vag" & tan.br.dr %in% c(NA,2,3,4,5))

## Set these variables as factors so the linear models treat them as such
meta.girl.vag$tan.br.dr <- factor(meta.girl.vag$tan.br.dr, ordered=TRUE) # set as ordered factor
meta.girl.vag$men.stat <- factor(meta.girl.vag$men.stat, levels=c("pre","post"))
meta.girl.vag$subject <- factor(meta.girl.vag$subject)
```

## Model comparisons for LAB
Now that we're all set up, we start testing regression models. The first is a very simple model containing four variables (Tanner stage, age, menarche status and subject):
```{r fit-lab-1}
## First, a very simple model with four variables and no interactions:
fit.lab.1 <- lm(LAB.logit ~ tan.br.dr + age.sampling + men.stat + subject, meta.girl.vag)
anova(fit.lab.1) 
summary(fit.lab.1)
```

From the first simple model, only men.stat is not significant. However, let's keep it for now while we test interactions. The next model is very complex with all 2-way, 3-way and 4-way interactions:
```{r fit-lab-2}
fit.lab.2 <- lm(LAB.logit ~ tan.br.dr + age.sampling + men.stat + subject + # single terms
                  tan.br.dr:age.sampling + tan.br.dr:men.stat + tan.br.dr:subject + # 2-way interactions
                  age.sampling:men.stat + age.sampling:subject + men.stat:subject + # 2-way interactions
                  tan.br.dr:age.sampling:men.stat + tan.br.dr:age.sampling:subject + # 3-way interactions
                  tan.br.dr:men.stat:subject + age.sampling:men.stat:subject + # 3-way interactions
                  tan.br.dr:age.sampling:men.stat:subject, # 4-way interaction 
                meta.girl.vag)
anova(fit.lab.2)
# summary(fit.lab.2) # this is very long

anova(fit.lab.1, fit.lab.2) # compare 2 to 1
```

Outcome of model 2 vs. 1: models are significantly different (p=0.0068), so keep more complex model 2 and reduce further in next step. Now we will drop the insignificant terms from fit.lab.2:
```{r fit-lab-3}
fit.lab.3 <- lm(LAB.logit ~ tan.br.dr + age.sampling + subject + # single terms
                  tan.br.dr:age.sampling + tan.br.dr:men.stat + tan.br.dr:subject + age.sampling:subject + # 2-way interactions
                  tan.br.dr:age.sampling:men.stat, # 3-way interaction
                meta.girl.vag)
anova(fit.lab.3)
# summary(fit.lab.3) # this is very long
anova(fit.lab.2, fit.lab.3) # compare 3 to 2
```

Outcome of model 3 vs. 2: models are not significantly different (p=0.2754), so keep simpler model 3. Next, try dropping the 3-way term from fit.lab.3:
```{r fit-lab-4}
fit.lab.4 <- lm(LAB.logit ~ tan.br.dr + age.sampling + subject + # single terms
                  tan.br.dr:age.sampling + tan.br.dr:men.stat + tan.br.dr:subject + age.sampling:subject, # 2-way interactions
                meta.girl.vag)
anova(fit.lab.4)
# summary(fit.lab.4) # this is very long
anova(fit.lab.3, fit.lab.4) # compare 4 to 3
```

Outcome of model 4 vs. 3: models are not significantly different (p=0.2233), so keep simpler model 4. At this point all terms in model 4 are significant, but let's just try dropping the 2-way interaction with the highest p-value, tan.br.dr:men.stat :
```{r fit-lab-5}
fit.lab.5 <- lm(LAB.logit ~ tan.br.dr + age.sampling + subject + # single terms
                  tan.br.dr:age.sampling + tan.br.dr:subject + age.sampling:subject, # 2-way interactions
                meta.girl.vag)
anova(fit.lab.5)
# summary(fit.lab.5) # this is very long
anova(fit.lab.4, fit.lab.5) # compare 5 to 4
```

Outcome of model 5 vs. 4: models are not significantly different (p=0.4358), so keep simpler model 5. Let's try dropping tan.br.dr:subject now:
```{r fit-lab-6}
fit.lab.6 <- lm(LAB.logit ~ tan.br.dr + age.sampling + subject + # single terms
                  tan.br.dr:age.sampling + age.sampling:subject, # 2-way interactions
                meta.girl.vag)
anova(fit.lab.6)
# summary(fit.lab.6) # this is fairly long
anova(fit.lab.5, fit.lab.6) # compare 6 to 4
```

Outcome of model 6 vs. 5: models are significantly different (p=0.0087), so we'll stick with slightly more complex model 5. Model 5 is considerably more complex than our simple model 1, but let's compare them to see how different they are:
```{r fit-lab-1vs4}
anova(fit.lab.1, fit.lab.5)
```

Model 5 is significantly different from model 1 (p=0.0004). Final decision: keep model 5, which includes tan.br.dr, age.sampling, subject, and three 2-way interaction terms as predictors of LAB. View the summary of model 5, then output the ANOVA table:
```{r fit-lab-final}
summary(fit.lab.5)
write.csv(anova(fit.lab.5), file="anova-lab-best-lm.csv")
```

## Model comparisons for vaginal pH

Now we use the same approach to find a multiple linear regression model for vaginal pH. First, a very simple model with four variables and no interactions:
```{r fit-ph-1}
fit.ph.1 <- lm(ph ~ tan.br.dr + age.sampling + men.stat + subject, meta.girl.vag)
anova(fit.ph.1) 
summary(fit.ph.1)
```

Once again, only men.stat is not significant. However, let's keep it for now while we test interactions. The next is a complex model with all 2-way, 3-way and 4-way interactions:
```{r fit-ph-2}
fit.ph.2 <- lm(ph ~ tan.br.dr + age.sampling + men.stat + subject + # single terms
                  tan.br.dr:age.sampling + tan.br.dr:men.stat + tan.br.dr:subject + # 2-way interactions
                  age.sampling:men.stat + age.sampling:subject + men.stat:subject + # 2-way interactions
                  tan.br.dr:age.sampling:men.stat + tan.br.dr:age.sampling:subject + # 3-way interactions
                  tan.br.dr:men.stat:subject + age.sampling:men.stat:subject + # 3-way interactions
                  tan.br.dr:age.sampling:men.stat:subject, # 4-way interaction 
                meta.girl.vag)
anova(fit.ph.2)
# summary(fit.ph.2) # this is very long
anova(fit.ph.1, fit.ph.2) # compare 2 to 1
```

Outcome of model 2 vs. 1: models are significantly different (p=0.0154), so keep more complex model 2 and reduce further. Next, drop the insignificant terms from fit.ph.2:
```{r fit-ph-3}
fit.ph.3 <- lm(ph ~ tan.br.dr + age.sampling + subject + # single terms
                  tan.br.dr:men.stat + tan.br.dr:subject + age.sampling:subject + men.stat:subject, # 2-way interactions 
                meta.girl.vag)
anova(fit.ph.3)
# summary(fit.ph.3) # this is very long
anova(fit.ph.2, fit.ph.3) # compare 3 to 2
```

Outcome of model 3 vs. 2: models 2 and 3 are not significantly different (p=0.0691), so keep simpler model 3. Next, drop the insignificant terms from fit.ph.3:
```{r fit-ph-4}
fit.ph.4 <- lm(ph ~ tan.br.dr + age.sampling + subject + # single terms
                  tan.br.dr:men.stat + tan.br.dr:subject, # 2-way interactions 
                meta.girl.vag)
anova(fit.ph.4)
# summary(fit.ph.4) # this is fairly long
anova(fit.ph.3, fit.ph.4) # compare 4 to 3
```

Outcome of model 4 vs. 3: models 3 and 4 are not significantly different (p=0.1026), so keep simpler model 4. Next, drop the insignificant terms from fit.ph.4 (leaving three single terms only):
```{r fit-ph-5}
fit.ph.5 <- lm(ph ~ tan.br.dr + age.sampling + subject, meta.girl.vag)
anova(fit.ph.5)
summary(fit.ph.5)
anova(fit.ph.4, fit.ph.5) # compare 5 to 4
```

Outcome of model 5 vs. 4: models 4 and 5 are not significantly different (p=0.1036), so keep simpler model 5. All three factors in model 5 are highly significant, so it doesn't make sense to try taking another term out. Finally, compare model 5 to the first simple model 1:
```{r fit-ph-5vs1}
anova(fit.ph.1, fit.ph.5)
```

Final decision: models 1 and 5 are not significantly different (p=0.4123), so keep simpler model 5. Output the ANOVA table:
```{r fit-ph-final}
write.csv(anova(fit.ph.5), file="table-s7-anova-ph-best-lm.csv")
```

## Recap of regression model comparisons
To recap, we selected the following as optimal multiple linear regression models for LAB (logit-transformed) and vaginal pH:

* LAB.logit ~ tan.br.dr + age.sampling + subject + tan.br.dr:age.sampling + tan.br.dr:men.stat + tan.br.dr:subject + age.sampling:subject
* ph ~ tan.br.dr + age.sampling + subject

***
# Part III. Plotting trends in lactic acid bacteria and vaginal pH

## Trends in LAB associated with participant metadata
```{r lab-trends}
## Boxplot of LAB.logit ~ Tanner breast (significant in linear model)
gg.lab.logit.tb <- ggplot(subset(meta.girl.vag, tan.br.dr %in% c(2,3,4,5)), 
                          aes(y=LAB.logit, x=tan.br.dr, fill=tan.br.dr))
box.lab.logit.tb <- gg.lab.logit.tb + 
  geom_boxplot(width=0.5, outlier.shape=1) +
  scale_fill_manual(values=col.tanner[2:5], name="Tanner\nBreast") +
  guides(fill=FALSE) +
  xlab("Tanner breast") +
  ylab("Logit-transformed LAB proportion") +
  theme_cust

print(box.lab.logit.tb)

## Boxplot of LAB.logit ~ menarche status (not significant in linear model)
gg.lab.logit.ms <- ggplot(subset(meta.girl.vag, men.stat %in% c("pre","post")), 
                          aes(y=LAB.logit, x=men.stat, fill=men.stat))
box.lab.logit.ms <- gg.lab.logit.ms + 
  geom_boxplot(width=0.5, outlier.shape=1) +
  scale_fill_manual(values=col.men.stat[c(3,2)], name="Menarche\nStatus") +
  guides(fill=FALSE) +
  xlab("Menarche status") +
  ylab("Logit-transformed LAB proportion") +
  theme_cust

print(box.lab.logit.ms)

## Scatterplot of LAB.logit ~ age (significant in linear model)
gg.lab.logit.age <- ggplot(meta.girl.vag, 
                           aes(y=LAB.logit, x=age.sampling))
scatter.lab.logit.age <- gg.lab.logit.age + 
  geom_point(size=2, alpha=0.7) +
  xlab("Age (y)") +
  ylab("Logit-transformed LAB proportion") +
  theme_cust

print(scatter.lab.logit.age)

## Boxplot of LAB.logit ~ subject (significant in linear model)
gg.lab.logit.sub <- ggplot(meta.girl.vag, 
                           aes(y=LAB.logit, x=subject, fill=subject))
box.lab.logit.sub <- gg.lab.logit.sub + 
  geom_boxplot(width=0.5, outlier.shape=1) +
  scale_fill_manual(values=col.gm.pair[1:31], name="Subject") +
  guides(fill=FALSE) +
  xlab("Subject") +
  ylab("Logit-transformed LAB proportion") +
  theme_cust

print(box.lab.logit.sub)
```

## Trends in vaginal pH associated with participant metadata
```{r ph-trends}
## Boxplot of pH ~ Tanner breast (significant in linear model)
gg.ph.tb <- ggplot(subset(meta.girl.vag, tan.br.dr %in% c(2,3,4,5)), 
                   aes(y=ph, x=tan.br.dr, fill=tan.br.dr))
box.ph.tb <- gg.ph.tb + 
  geom_boxplot(width=0.5, outlier.shape=1) +
  scale_fill_manual(values=col.tanner[2:5], name="Tanner\nBreast") +
  guides(fill=FALSE) +
  xlab("Tanner breast") +
  ylab("Vaginal pH") +
  theme_cust

print(box.ph.tb)

## Boxplot of pH ~ menarche status (not significant in linear model)
gg.ph.ms <- ggplot(subset(meta.girl.vag, men.stat %in% c("pre","post")), 
                   aes(y=ph, x=men.stat, fill=men.stat))
box.ph.ms <- gg.ph.ms + 
  geom_boxplot(width=0.5, outlier.shape=1) +
  scale_fill_manual(values=col.men.stat[c(3,2)], name="Menarche\nStatus") +
  guides(fill=FALSE) +
  xlab("Menarche status") +
  ylab("Vaginal pH") +
  theme_cust

print(box.ph.ms)

## Scatterplot of pH ~ age (significant in linear model)
gg.ph.age <- ggplot(meta.girl.vag, 
                    aes(y=ph, x=age.sampling))
scatter.ph.age <- gg.ph.age + 
  geom_point(size=2, alpha=0.7) +
  xlab("Age (y)") +
  ylab("Vaginal pH") +
  theme_cust

print(scatter.ph.age)

## Boxplot of ph ~ subject (significant in linear model)
gg.ph.sub <- ggplot(meta.girl.vag, 
                    aes(y=ph, x=subject, fill=subject))
box.ph.sub <- gg.ph.sub + 
  geom_boxplot(width=0.5, outlier.shape=1) +
  scale_fill_manual(values=col.gm.pair[1:31], name="Subject") +
  guides(fill=FALSE) +
  xlab("Subject") +
  ylab("Vaginal pH") +
  theme_cust

print(box.ph.sub)
```

## Figure 6. Trends in relative abundance of lactic acid bacteria and vaginal pH in relation to pubertal development and menarche status.
198 vaginal swabs were collected from 31 girls over time. Upper and lower panels show box plots of (a) the logit-transformed proportion of lactic acid bacteria (LAB; includes _Lactobacillus_, _Streptococcus_, _Aerococcus_ and _Facklamia_) and (b) vaginal pH. In the left column, box plots show the relationship to Tanner breast score (191 data points for LAB and 130 data points for pH out of 198 total samples). Box plots in the right column show the relationship to menarche status (197 data points for LAB and 130 data points for pH). In each plot the rectangular box represents the interquartile range, the whiskers represent the upper and lower quartiles, the horizontal line represents the median, and open circles represent outliers.

```{r fig-6-lab-ph-trends, fig.width=6, fig.height=8}
## Make compound figure for manuscript with LAB.logit panels on top, pH on bottom
fig.lab.logit.ph.1 <- box.lab.logit.tb +
  ggtitle("a") +
  theme(plot.title=element_text(size=22, hjust=-0.15, vjust=1.5))
fig.lab.logit.ph.2 <- box.lab.logit.ms +
  ggtitle("") +
  ylab("") +
  theme(plot.title=element_text(size=22, hjust=-0.15, vjust=1.5))
fig.lab.logit.ph.3 <- box.ph.tb +
  ggtitle("b") +
  theme(plot.title=element_text(size=22, hjust=-0.15, vjust=1.5))
fig.lab.logit.ph.4 <- box.ph.ms +
  ggtitle("") +
  ylab("") +
  theme(plot.title=element_text(size=22, hjust=-0.15, vjust=1.5))

# pdf("fig-6-lab-logit-ph-trends.pdf", width=6, height=8)
multiplot(fig.lab.logit.ph.1, fig.lab.logit.ph.2,  
          fig.lab.logit.ph.3, fig.lab.logit.ph.4,  
          layout=matrix(c(1,1,1,2,2,3,3,3,4,4), ncol=5, byrow=TRUE))
dev.off()
```

## Discordance between high LAB and low pH

An interesting observation was that not all samples with high proportions of LAB were associated with a low pH. We hypothesized this could be due to lower total bacterial counts in the vaginas of adolescent girls. We performed a coarse test of that hypothesis (see Supplemental File 5, 'adolescent-supp-05.Rmd) but were unable to detect a significant difference in estimated number of 16S rRNA gene copies.

## Figure 7. Relationship between proportion of lactic acid bacteria and pH in vaginal samples collected from girls.
197 vaginal microbiota samples from girls are plotted to show the relationship between the proportion of lactic acid bacteria (LAB; includes _Lactobacillus_, _Streptococcus_, _Aerococcus_ and _Facklamia_) on the x-axis and vaginal pH on the y-axis. Menarche status is either premenarcheal (open circles) or postmenarcheal (filled circles). Points are color-coded to indicate Tanner breast stage as indicated by the legend at right (NA values are colored gray). Points are slightly jittered to decrease crowding around similar values, but not so much as to distort interpretation of the data. The magenta dashed line at vaginal pH 4.5 represents the upper range of the traditional ‘hallmark’ healthy vaginal pH of 4.0-4.5.

```{r fig-7-lab-ph, fig.width=8, fig.height=6}
## LAB vs. ph
gg.lab.ph <- ggplot(subset(meta, type=="girl" & site=="vag"), 
                    aes(x=LAB, y=ph, col=factor(tan.br.dr), shape=men.stat))

gg.lab.ph + geom_hline(yintercept=4.5, linetype=2, color="#8B1C62", alpha=0.8) +
  geom_jitter(position=position_jitter(width = 0.01, height = 0.05), size=2.5, alpha=0.8) +
  scale_color_manual(values=col.tanner, name="Tanner\nBreast", na.value="gray70") +
  scale_shape_manual(values=c(16,1), name="Menarche\nStatus", breaks=c("pre", "post")) +
  xlab("Proportion of LAB") +
  ylab("Vaginal pH") +
  theme_cust

# Uncomment line below to save as PDF
# ggsave("fig-7-lab-ph-scatterplot.pdf", width=8, height=6, units="in")
```

End of Supplemental File 3. For a description of qPCR performed to test whether low-pH samples had more bacterial 16S gene copies than high-pH samples, see *Supplemental File 5. Comparison of low-pH vs. high-pH vaginal microbiota with pan-bacterial 16S rRNA qPCR assay.* For more exploration of the vulvar microbiota, see *Supplemental File 6. Comparisons of vaginal and vulvar microbiota of girls.*